crates/services/compression/src/temporal_registry.rs

```diff
@@ -53,7 +62,48 @@ pub struct CompressionStorageWrapper<'a, CS> {
 /// necessary metadata needed to perform compression
 pub struct CompressionContext<'a, CS> {
     pub(crate) compression_storage: CompressionStorageWrapper<'a, CS>,
-    pub(crate) block_events: &'a [Event],
+    tx_pointers: HashMap<UtxoId, TxPointer>,
+}
+
+impl<'a, CS> CompressionContext<'a, CS> {
+    /// Creates a new compression context
+    pub fn create_from_block(
+        storage_tx: &'a mut StorageTransaction<CS>,
+        block: &'a Block,
+        chain_id: ChainId,
+    ) -> anyhow::Result<Self> {
+        let mut tx_pointers = HashMap::new();
+        for (tx_index, tx) in block.transactions().iter().enumerate() {
+            let tx_index = u16::try_from(tx_index)
+                .map_err(|_| anyhow::anyhow!("Transaction index exceeds u16 limit"))?;
+            let tx_id = tx.id(&chain_id);
+            for input in tx.inputs().iter() {
+                if let fuel_core_types::fuel_tx::Input::CoinPredicate(coin) = input {
+                    let utxo_id = coin.utxo_id;
+                    let tx_pointer = coin.tx_pointer;
+                    tx_pointers.insert(utxo_id, tx_pointer);
+                } else if let fuel_core_types::fuel_tx::Input::CoinSigned(coin) = input {
+                    let utxo_id = coin.utxo_id;
+                    let tx_pointer = coin.tx_pointer;
+                    tx_pointers.insert(utxo_id, tx_pointer);
+                }
+            }
+            for (index, output) in tx.outputs().iter().enumerate() {
+                let index = u16::try_from(index)
+                    .map_err(|_| anyhow::anyhow!("Output index exceeds u16 limit"))?;
+                if let fuel_core_types::fuel_tx::Output::Coin { .. } = output {
```

Comment:
I think `Variable` and `Change` output must be used as well, because it is possible that next transactions will use these coins.

Reply:
Added followup issue: https://github.com/FuelLabs/fuel-core/issues/3073